### PR TITLE
NUTCH-2780 : Upgrade index-solr to use Solr 8.5.1

### PR DESCRIPTION
**Pull request by balashashanka**
Upgraded the solr client to 8.5.1
Registered all the dependencies in plugin.xml

### CODE REVIEW

File: src/plugin/indexer-solr/ivy.xml [hunk 1]
```diff
@@ -36,9 +36,9 @@
   </publications>
 
   <dependencies>
-    <dependency org="org.apache.solr" name="solr-solrj" rev="7.3.1"/>
-    <dependency org="org.apache.httpcomponents" name="httpcore" rev="4.4.6" conf="*->default"/>
-    <dependency org="org.apache.httpcomponents" name="httpmime" rev="4.5.3" conf="*->default"/>
+    <dependency org="org.apache.solr" name="solr-solrj" rev="8.5.1"/>
+    <dependency org="org.apache.httpcomponents" name="httpcore" rev="4.4.12" conf="*->default"/>
+    <dependency org="org.apache.httpcomponents" name="httpmime" rev="4.5.10" conf="*->default"/>
```
[the remark this answers is not in the document]
Comment (balashashanka) [reply src/plugin/indexer-solr/ivy.xml:41]: Hi @sebastian-nagel , i tried fixing this but with no positive results. Could you tell me how you fixed it?

**Comment by sebastian-nagel** [comment]
Hi @balashashanka, you'll find all fixes in c6646fd. Ok, I forgot to mention that the slf4j-api dependency needs also be removed from the plugin.xml. Don't know whether the ivy version makes a difference, but we use ivy 2.4.0 in core, so we should use it also for the plugins. For testing, I've installed Solr 8.5.1 as described in the [Nutch tutorial](https://cwiki.apache.org/confluence/display/NUTCH/NutchTutorial#NutchTutorial-SetupSolrforsearch) and used the tool indexchecker.

**Comment by balashashanka** [comment]
Removing the sl4j-api fixed it for me thanks. I've added the howto_upgrade_solr.txt with the reference from the elastic [howto](https://github.com/apache/nutch/blob/master/src/plugin/indexer-elastic/howto_upgrade_es.txt). Will test the changes and commit.

**Comment by sebastian-nagel** [comment]
Excellent! Indexing works now out of the box. I'll commit soon. Thanks, @balashashanka!

**Pull request merged**